fix(Flow): prevent calling onFlowChange on initial render

plagoa opened the pull request.

plagoa commented on packages/lab/src/components/Flow/DroppableFlow.tsx:
```diff
@@ -130,6 +130,7 @@ export const HvDroppableFlow = ({
   ...others
 }: HvDroppableFlowProps) => {
   const { classes, cx } = useClasses(classesProp);
+  const [initialRender, setInitialRender] = useState(true);
```
Oh perfect, I'll use that. Thanks!

MEsteves22 commented on packages/lab/src/components/Flow/DroppableFlow.tsx:
```diff
@@ -224,10 +225,14 @@ export const HvDroppableFlow = ({
 
-      handleFlowChange(nds, edges);
-      onNodesChangeProp?.(changes);
+      if (!initialRender.current) {
```
This way, if we have an empty canvas and add one node to it, the `onFlowChange` won't be triggered. I believe setting the `ref` when the component's mount as before would solve this.

plagoa replied:
Yeah, I'll keep the `ref` but revert back to the original solution. The `useIsMounted` hook doesn't work in what we're trying to achieve here.

MEsteves22 commented on packages/lab/src/components/Flow/DroppableFlow.tsx:
```diff
@@ -144,6 +145,10 @@ export const HvDroppableFlow = ({
     id: elementId,
   });
 
+  useEffect(() => {
```
It doesn't seem to be working when testing the "Initial state" sample. The `ref` is set before `onNodesChange` is triggered. 

I believe one way of solving this is using the [`onInit`](https://reactflow.dev/api-reference/react-flow#on-init) callback:

```
const handleInit = useCallback(= (i) => {
   initialRender.current = false;
   onInitProp?.(i) // Extracted from props passed by the used
}, [onInitProp])

<ReactFlow
   (...)
   onInit={handleInit}
>
{children}
</ReactFlow>
```

plagoa commented:
@MEsteves22 @zettca as discussed I removed the "first render" logic from the component as it looked a bit sketchy and actually prevented the user from being notified of that first render. I updated the "InitialState" sample to include the behavior we were looking for but have it on the application side.

MEsteves22 commented:
> @MEsteves22 @zettca as discussed I removed the "first render" logic from the component as it looked a bit sketchy and actually prevented the user from being notified of that first render. I updated the "InitialState" sample to include the behavior we were looking for but have it on the application side.

Great 🙏 We just need to remove the code from the droppable flow

zettca reviewed (approved):
👍
